### PR TITLE
NoteSync hybrid sample app

### PR DESCRIPTION
**Pull request by wmathurin**
Will remove mobile2.t client id key once we have a sandbox org with content notes enabled

### CODE REVIEW

File: samples/notesync/NoteSync.html [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+    render: function(eventName) {
+        $(this.el).html(this.template());
+        $(".search-key", this.el).val(this.model.getCriteria());
+        this.listView.setElement($("ul", this.el)).render();
+        return this;
+    },
+
+    syncDown: function() {
+        var target = {type:"custom", androidImpl:"com.salesforce.samples.notesync.ContentSoqlSyncTarget", query:"SELECT " + this.fieldlist.join(",") + " FROM ContentNote LIMIT 1000"};
```
Comment (wmathurin) [inline samples/notesync/NoteSync.html]: Custom target

File: samples/notesync/NoteSync.html [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    template: _.template($("#note-page").html()),
+
+    events: {
+        "change": "change",
+        "click .save": "save"
+    },
+
+    render: function(eventName) {
+        var templateData = _.extend({ContentDecoded: atob(this.model.get("Content"))}, this.model.toJSON());
```
Comment (wmathurin) [inline samples/notesync/NoteSync.html]: base64 decoding at view time

File: samples/notesync/NoteSync.html [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        $(this.el).html(this.template(templateData));
+        return this;
+    },
+
+    change: function(event) {
+        // apply change to model
+        var target = event.target;
+        if (target.name == "ContentDecoded") {
+            this.model.set("Content", btoa(target.value));
```
Comment (wmathurin) [inline samples/notesync/NoteSync.html]: base64 encoding at save time

File: samples/notesync/bootconfig.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "remoteAccessConsumerKey": "3MVG9Iu66FKeHhIMmUBkEV8o6PfICh5xUnN89HyifhHcvSSBM3WUYyG9buZBezdrhdYilqlXHHbqamC5MqaKP", // mobile2
```
Comment (wmathurin) [inline samples/notesync/bootconfig.json]: Key that will soon be removed

**Comment by wmathurin** [comment]
That was already reviewed when I sent a pull request from wmathurin notesync to forcedotcom notesync

**Comment by bhariharan** [comment]
Should this go into the ```notesync``` branch?

**Comment by wmathurin** [comment]
It's already in the notesync branch.
That's the merge from notesync to unstable.
Except I did in my fork first.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**